src/main/java/org/apache/commons/imaging/formats/gif/GifImageParser.java

```diff
@@ -407,7 +407,7 @@ public byte[] getIccProfileBytes(final ByteSource byteSource, final GifImagingPa
     @Override
     public ImageInfo getImageInfo(final ByteSource byteSource, final GifImagingParameters params)
             throws ImagingException, IOException {
-        final GifImageContents blocks = readFile(byteSource, false);
+        final GifImageContents blocks = readFile(byteSource, true);
```

Comment:
Maybe this `true`/`false` (`stopBeforeImageData`) should instead be retrieved from the `GitImagingParameters` object?

Reply:
That's a good point!

I wonder if this data is hard-coded to be skipped it will be unavailable for other API calls once the object is constructed. That would be bad. But if it can be paramerized, then the client call.site can decide.

Reply:
**Note**: All current callers of `org.apache.commons.imaging.formats.gif.GifImageParser.readFile(ByteSource, boolean)` pass false for `stopBeforeImageData`.

Would it make for all call sites to pick up this value from `GifImagingParameters`? Or, would this cause some use-cases to break. We will need to make sure to unit test whatever code we change or add.

Reply:
`stopReadingBeforeImageData` or `skipImageData` works.